godtools/Views/TractElements/BaseTractElement.swift

```diff
@@ -0,0 +1,175 @@
+//
+//  BaseTractElement.swift
+//  godtools
+//
+//  Created by Ryan Carlson on 4/26/17.
+//  Copyright © 2017 Cru. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SWXMLHash
+
+class BaseTractElement: UIView {
+    struct Standards {
+        static let xMargin = CGFloat(8.0)
+        static let yMargin = CGFloat(8.0)
+        static let xPadding = CGFloat(0.0)
+        static let yPadding = CGFloat(0.0)
+        
+        static let screenWidth = UIScreen.main.bounds.size.width
+        static let textContentWidth = UIScreen.main.bounds.size.width - xMargin * CGFloat(2)
+    }
+    
+    weak var parent: BaseTractElement?
+    var elements:[BaseTractElement]?
+    var yStartPosition: CGFloat = 0.0
+    var maxHeight: CGFloat = 0.0
+    private var _height: CGFloat = 0.0
+    var height: CGFloat {
+        get {
+            return _height
+        }
+        set {
+            _height = newValue
+        }
+    }
+    var width: CGFloat {
+        if (self.parent != nil) {
+            return (self.parent?.width)!
+        } else {
+            return BaseTractElement.Standards.screenWidth
+        }
+    }
+    var horizontalContainer: Bool {
```

Comment:
Most of the times will be false, but I need for some view (and maybe another views) in the future to be true, so in the children class I override this property and thats why I need a return, also, because it is easier to do that than setting `self.horizontalContainer = true` on the children class, because in order to do that it will require to override another method (because of the moment when the property is being used)